modules/material-management/src/multi_keyring.ts

```diff
@@ -113,19 +113,29 @@ function buildPrivateOnDecrypt<S extends SupportedAlgorithmSuites> () {
     const children = this.children.slice()
     if (this.generator) children.unshift(this.generator)
 
+    let keyringError = false
+
     for (const keyring of children) {
     /* Check for early return (Postcondition): Do not attempt to decrypt once I have a valid key. */
       if (material.hasValidKey()) return material
 
       try {
         await keyring.onDecrypt(material, encryptedDataKeys)
       } catch (e) {
-      // there should be some debug here?  or wrap?
       // Failures onDecrypt should not short-circuit the process
       // If the caller does not have access they may have access
       // through another Keyring.
+        keyringError = true
       }
     }
+
+    /* Postcondition: Need either a valid data key or no errors.
```

Comment:
This logic is in line with what is currently in the spec PR. However, I either want someone familiar with the C multikeyring implementation to either give this logic the OK, or give the current wording in the spec the OK, before I approve this. I will tag someone in the spec CR on this issue.

Reply:
Thanks :)